src/main/kotlin/org/koitharu/kotatsu/parsers/site/TuMangaOnlineParser.kt

```diff
@@ -0,0 +1,382 @@
+package org.koitharu.kotatsu.parsers.site
+
+import android.os.SystemClock
+import okhttp3.FormBody
+import okhttp3.Headers
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import org.koitharu.kotatsu.parsers.MangaLoaderContext
+import org.koitharu.kotatsu.parsers.MangaSourceParser
+import org.koitharu.kotatsu.parsers.PagedMangaParser
+import org.koitharu.kotatsu.parsers.config.ConfigKey
+import org.koitharu.kotatsu.parsers.model.Manga
+import org.koitharu.kotatsu.parsers.model.MangaChapter
+import org.koitharu.kotatsu.parsers.model.MangaPage
+import org.koitharu.kotatsu.parsers.model.MangaSource
+import org.koitharu.kotatsu.parsers.model.MangaState
+import org.koitharu.kotatsu.parsers.model.MangaTag
+import org.koitharu.kotatsu.parsers.model.RATING_UNKNOWN
+import org.koitharu.kotatsu.parsers.model.SortOrder
+import org.koitharu.kotatsu.parsers.util.attrAsAbsoluteUrlOrNull
+import org.koitharu.kotatsu.parsers.util.attrAsRelativeUrl
+import org.koitharu.kotatsu.parsers.util.domain
+import org.koitharu.kotatsu.parsers.util.generateUid
+import org.koitharu.kotatsu.parsers.util.host
+import org.koitharu.kotatsu.parsers.util.parseFailed
+import org.koitharu.kotatsu.parsers.util.parseHtml
+import org.koitharu.kotatsu.parsers.util.selectFirstOrThrow
+import org.koitharu.kotatsu.parsers.util.toAbsoluteUrl
+import java.io.IOException
+import java.text.SimpleDateFormat
+import java.util.EnumSet
+import java.util.Locale
+import java.util.concurrent.Semaphore
+import java.util.concurrent.TimeUnit
+
+@MangaSourceParser("TUMANGAONLINE", "TuMangaOnline", "es")
+class TuMangaOnlineParser(context: MangaLoaderContext) : PagedMangaParser (
+	context,
+	source = MangaSource.TUMANGAONLINE,
+	pageSize = 24,
+), Interceptor {
+
+	private val client = OkHttpClient.Builder().build()
+
+	override val headers: Headers
+		get() = super.headers.newBuilder()
+			.set("Referer","https://$domain/")
+			.build()
+
+	override val configKeyDomain: ConfigKey.Domain
+		get() = ConfigKey.Domain("lectortmo.com")
+
+	override val sortOrders: Set<SortOrder>
+		get() = EnumSet.of(
+			SortOrder.NEWEST,
+			SortOrder.POPULARITY,
+		)
+
+	private val host = domain
+	private val permits = 10
+	private val period = 60L
+	private val unit = TimeUnit.SECONDS
+
+	private val requestQueue = ArrayDeque<Long>(permits)
+	private val rateLimitMillis = unit.toMillis(period)
+	private val fairLock = Semaphore(1, true)
+
+	override fun intercept(chain: Interceptor.Chain): Response {
```

Comment:
What's going on here?

Reply:
I've also come across sources that ask not to spam so it would be a good thing later to provide setup a rate limit easily.

Reply:
Ok I suppose it might be better to move this into a standalone Interceptor. I'll do it

Reply:
@Koitharu How do I add `RateLimitInterceptor` to the source? or will it be done later?